source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+```javascript
+cy.route2(/users\/\d+/, { id: 1, name: 'Phoebe' })
+```
+
+```javascript
+// Application Code
+$.get('https://localhost:7777/users/1337', (data) => {
+  console.log(data) // => {id: 1, name: "Phoebe"}
+})
+```
+
+### Response functions
+
+You can also use a function as a response which enables you to add logic and modify properties of the response. The function exposes a `req` object which contains the request as well as a way to `send` with a response.
+
+```javascript
+cy.route2('/login', (req) => {
+  req.send('Success')
+})
+```
+
+```javascript
+// reply with body and headers
+cy.route2('/login', (req) => {
+  req.send('Success', { 'x-new-headers': 'from-server' })
+})
+```
+
+```javascript
+// reply with an object containing more response details
+cy.route2('/login', (req) => {
+  req.send({
+    statusCode: 200,
+    body: JSON.stringify({
+      some: 'response'
+    }),
+    headers: {
+      'x-new-headers': 'from-server'
+    },
+    // If `destroySocket` is truthy, Cypress will destroy the connection to the
+    // browser and send no response. Useful for simulating a server that is not
+    // reachable. Must not be set in combination with other options.
+    destroySocket: false
+  })
+})
+```
+
+```javascript
+// continue the HTTP response to the browser, including any modifications made to `res`
+cy.route2('/login', (req) => {
+  req.send()
+})
+```
+
+### Matching requests and routes
+
+Any request that matches the `method` and `url` of a route will be responded to based on the configuration of that route.
+
+{% note bolt %}
+By default, all HTTP methods will be used used to match routes. If you want to stub a route with a specific HTTP method such as `POST` then you {% urlHash 'must be explicit about the method' Arguments %}.
+{% endnote %}
+
+### Specify the method
+
+The below example matches all `DELETE` requests to "/users" and stubs a response with an empty JSON object.
+
+```javascript
+cy.route2('DELETE', '**/users/*', {})
+```
+
+### Making multiple requests to the same route
+
+You can test a route multiple times with unique response objects by using {% url 'aliases' variables-and-aliases#Aliases %} and {% url '`cy.wait()`' wait %}. Each time we use `cy.wait()` for an alias, Cypress waits for the next nth matching request.
+
+```javascript
+cy.route2('/beetles', []).as('getBeetles')
+cy.get('#search').type('Weevil')
+
+// wait for the first response to finish
+cy.wait('@getBeetles')
+
+// the results should be empty because we
+// responded with an empty array first
+cy.get('#beetle-results').should('be.empty')
+
+// now re-define the /beetles response
+cy.route2('/beetles', [{ name: 'Geotrupidae' }])
+
+cy.get('#search').type('Geotrupidae')
+
+// now when we wait for 'getBeetles' again, Cypress will
+// automatically know to wait for the 2nd response
+cy.wait('@getBeetles')
+
+// we responded with 1 beetle item so now we should
+// have one result
+cy.get('#beetle-results').should('have.length', 1)
+```
```

Comment:
might need to verify this example, not sure stubbing with an array like this will work, could be a potential bug

Reply:
I've updated the example to show multiple routes, but could use some feedback on whether a more practical example exists!